Source list MUI table

guynir42 opened the pull request:
Turn the table in the sources page to a MUI table. 

The search box will come in a separate PR. 

![source_list_table](https://user-images.githubusercontent.com/37179063/97081524-abefc180-160b-11eb-907f-2ca9ae1ed925.gif)

guynir42 commented:
I think the test failures are unrelated...

dannygoldstein reviewed (changes requested):
In the pullout, can we add 

- [ ] A spectrum plot (I added a VegaPlot for this that is being used to preview spectra in the UploadSpectrum form)

- [ ] An autoannotations table

guynir42 commented:
> In the pullout, can we add
> 
> * [ ]  A spectrum plot (I added a VegaPlot for this that is being used to preview spectra in the UploadSpectrum form)
> * [ ]  An autoannotations table

I added annotations, in the same style as the scanning page. 

As for spectra: I will open a separate PR for making a VegaPlot for spectra. 
There is an important question, though: there can be multiple spectra for the same source. Which one should we show?

acrellin commented on skyportal/handlers/api/source.py:
```diff
@@ -426,6 +426,7 @@ def get(self, obj_id=None):
             q = q.filter(Obj.within(other, radius))
         if start_date:
             start_date = arrow.get(start_date.strip())
+            print(start_date)
```
```suggestion
```

guynir42 replied:
Yeah that's what I was thinking, too. But I had the feeling there would be a lot of different stuff between the two pages, but it is starting to look like the two pages would be very similar.

acrellin reviewed (changes requested):
My main request for now is to use the source list table that we're already using in the group sources page. That might need to be slightly modified to be even further generalized to use here. I don't think we should be duplicating this much code.

dannygoldstein commented:
> > In the pullout, can we add
> > 
> > * [ ]   A spectrum plot (I added a VegaPlot for this that is being used to preview spectra in the UploadSpectrum form)
> > * [ ]   An autoannotations table
> 
> I added annotations, in the same style as the scanning page.
> 
> As for spectra: I will open a separate PR for making a VegaPlot for spectra.
> There is an important question, though: there can be multiple spectra for the same source. Which one should we show?

All.

guynir42 commented:
closes #1224

kmshin1397 reviewed (changes requested):
Couple of minor changes but looking good!

kmshin1397 commented on skyportal/tests/frontend/test_source_list.py:
```diff
@@ -1,119 +1,183 @@
+    )
+    assert status == 200
+
+    # check the classification shows up (it should not show up without a page refresh!)
+    try:
+        driver.wait_for_xpath(f"//*[text()[contains(., '{'Algol'}')]]", timeout=1)
+
+    except TimeoutException:
+        pass  # the classification should not appear on its own, so we ignore this error
```
```suggestion
    # check the classification doesn't shows up (it should not show up without a page refresh!)
    driver.wait_for_xpath_to_disappear(f"//*[text()[contains(., '{'Algol'}')]]", timeout=1)
```
Sorry @guynir42 one more thing. When you test for things to not be present, you should use the `wait_for_xpath_to_disappear` instead of the `try-catch` pattern I think - for readability, performance, and consistency.

guynir42 replied:
Thanks! I didn't know about that one...

kmshin1397 reviewed (commented):
Thanks for the changes! Sorry, just one last thing I missed last time.

kmshin1397 reviewed (approved):
Okay I think this looks good now. The test is still failing for me locally because we don't have server-side pagination yet and there are a lot of test sources to load before the timeout, but that can be worked on after this PR goes in.
